query/control/controller.go

```diff
@@ -45,6 +45,12 @@ import (
 // orgLabel is the metric label to use in the controller
 const orgLabel = "org"
 
+// MaxQueryQueueSize is the max allowable value for QueueSize
+// in controller config. The setting was found by trial-and-error,
+// it's the max size that can be passed to `makechan` for a channel
+// of struct pointers (also of int64s) without hitting a panic.
```

Comment:
Can you provide more info on the panic location so we can track future improvements to GoLang and possibly change this? Is there a known issue in GoLang we could reference, for instance?